File: 21558 - Nithin/week 1/844-backspace-string-compare/844-backspace-string-compare.cpp
class Solution {
public:
    bool backspaceCompare(string s, string t) {
        stack<char> stck1,stck2;
        bool flag = true;
        for(int i=0;i<s.size();i++){
            if(s[i] == '#'){
                if(!stck1.empty()){
                    stck1.pop();}
            }
            else
                stck1.push(s[i]);
        }
         for(int j=0;j<t.size();j++){
            if(t[j] == '#'){
                if(!stck2.empty()){
                    stck2.pop();}}
             else
                 stck2.push(t[j]);
        }
        if(stck1.size() != stck2.size())
            return false;
        
        while(stck1.empty() == false){
            if(stck1.top() == stck2.top()){
                stck1.pop();
                stck2.pop();
            }
            else{
                   flag = false;
                    break;
            }
        }
        return flag;
        
    }
};

File: 21561 - Suvam Tamang/Week 1/844_Backspace String Compare.java
class Solution {
    public boolean backspaceCompare(String s, String t) {
        Stack<Character> stack1 = new Stack<>();
        Stack<Character> stack2 = new Stack<>();
        
        for(char c1 : s.toCharArray()){
            if(c1 != '#'){
                stack1.push(c1);
            }
            else if(!stack1.empty()){
                stack1.pop();
            }
        }
        
        for(char c2 : t.toCharArray()){
            if(c2 != '#'){
                stack2.push(c2);
            }
            else if(!stack2.empty()){
                stack2.pop();
            }
        }

        return stack1.equals(stack2);
    }
}
